src/os.cc

```diff
@@ -218,7 +218,7 @@ int buffered_file::fileno() const {
 
 #if FMT_USE_FCNTL
 file::file(cstring_view path, int oflag) {
-  int mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
+  unsigned int mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
```

Comment:
Wouldn't warning disappear if `constexpr` is used, so that the compiler will have to substitute the value and not being afraid of narrowing. Then it could be `constexpr auto` I guess.

Reply:
I don't think `constexpr` is relevant here and `auto` will just result in `int`. Note that the warning is not on this line.